dlrs/main/classes/RollupSummariesSelector.cls

```diff
@@ -49,16 +54,42 @@ public class RollupSummariesSelector {
       enforceSecurity,
       forEdit
     );
+
+    String cacheKey = String.format(
```

Comment:
The data cache key is the security/usage parameters provided in the constructor.

Reply:
Which results in something like `false|true` - for readability's sake, what do you think about `String.valueOf(cacheKeyEnum)`, `cacheKeyEnum` being a readable representation of the four states
```
SelectedForReadWithSecurity
SelectedForEditWithSecurity
SelectedForReadWithoutSecurity
SelectedForEditWithoutSecurity
```

Reply:
I thought about that. I didn't make reading the cache key a priority because it was never exposed anywhere. I didn't think anyone would need it for troubleshooting or comprehending the code.

I wouldn't want to change the constructor interface so the caller passed in an Enum, I think that would be a very large change. Converting the current passed in variables to an enum would just be a multi-step `if` chain that didn't feel like was adding value.
I think if we wanted to build a string based on constructor variables that could make sense, rather that deriving one from the string values directly. That could be done easily with two ternary statements.

I'll put through a change so we can see if it feels better.

Reply:
One additional thought on this. Maybe the concern is more about tests that need to pre-fill caches, in which case they need to know the cache key so a human will have to read/set the data.
What if we provided a `@TestVisible` method using the recommended Enum to set the cache value. It would mean we'd have to go ahead and add an internal method with the `if` chain that translates the constructor arguments to an enum, I'd probably also expose an additional constructor override that allows you to pass the enum directly and we'd slowly convert the codebase to use that. This would be a much longer play but the test classes that need to set the cache directly might really appreciate it down the road.

I'd probably provide a public method to clear the cache as well, no parameters though, just burn all the caches.

I'm also thinking about the ideal future where test classes can set RollupSummary config state without inserting records, this will be necessary when we eventually want to deprecate the Custom Object which is still used heavily in our tests.